File: java-webflux-r2dbc/src/main/java/com/github/anddd7/entity/Product.java
package com.github.anddd7.entity;

import java.math.BigDecimal;
import org.springframework.data.annotation.Id;
import org.springframework.data.relational.core.mapping.Table;

@Table("products")
public class Product {

  @Id
  private Integer id;
  private String name;
  private BigDecimal price;

  public Integer getId() {
    return id;
  }

  public void setId(Integer id) {
    this.id = id;
  }

  public String getName() {
    return name;
  }

  public void setName(String name) {
    this.name = name;
  }

  public BigDecimal getPrice() {
    return price;
  }

  public void setPrice(BigDecimal price) {
    this.price = price;
  }
}

File: kotlin-webflux-r2dbc/src/main/kotlin/com/github/anddd7/dto/Product.kt
package com.github.anddd7.dto

import org.springframework.data.annotation.Id
import org.springframework.data.relational.core.mapping.Table
import java.math.BigDecimal

@Table("products")
data class Product(
    @Id
    val id: Int = 0,
    val name: String,
    val price: BigDecimal
)

data class ProductStockDTO(
    val id: Int = 0,
    val name: String,
    val price: BigDecimal,
    val stock: BigDecimal
) {
    constructor(product: Product, stock: BigDecimal) : this(product.id, product.name, product.price, stock)
}
